### PR TITLE
Terminal layer

### PR DESCRIPTION
**Pull request by Alex-H1**
added outline for terminal layer

### CODE REVIEW

**Comment by ftranlaba** [comment]
Hi Alexander! It looks like a great start. However, there are unnecessary commits. There shouldn't be any .class files or any other unintentional files. You can delete the unnecessary files and recommit or undo commits and recommit.

**Comment by Alex-H1** [comment]
added some functionality now just need to update db

File: src/main/java/terminallayer/util/TerminalUtil.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                try {
+                    throw new InvalidTypeException("Invalid Format");
+                } catch (InvalidTypeException e) {
+                    LOGGER.error(e);
+                }
+            }
+        }).thenRunAsync(() -> {
+            LOGGER.info("Please Enter CVC: ");
+            String cvc = scan.nextLine();
+            if (cvc.matches("[0-9]{4}")) {
```
Comment (Moussa-Mikhail) [inline src/main/java/terminallayer/util/TerminalUtil.java]: cvc is 3 numbers long.

File: src/main/java/terminallayer/util/TerminalUtil.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+public class TerminalUtil {
+
+    private static final Logger LOGGER = LogManager.getLogger("TESTLOGGER");
+    private static final Scanner scan = new Scanner(System.in);
+
+    public final static Account authUser() throws InvalidTypeException, ExecutionException, InterruptedException, SQLException {
+        AccountsDAO accountsDAO = new AccountsDAO();
+        Card card = new Card();
+        Account a = new Account();
```
Comment (Moussa-Mikhail) [inline src/main/java/terminallayer/util/TerminalUtil.java]: This variable isn't necessary.

File: src/main/java/terminallayer/TerminalMain.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        }).thenRunAsync(() -> {
+            LOGGER.info("Enter Address");
+            String address = scan.nextLine();
+            u.setAddress(address);
+        }).thenRunAsync(() -> {
+            LOGGER.info("Enter Phone Number: ");
+            String phoneNumber = scan.nextLine();
+            u.setPhoneNumber(phoneNumber);
+        });
+        u.setId(userIdCounter());
```
Comment (Moussa-Mikhail) [inline src/main/java/terminallayer/TerminalMain.java]: You should use the UserDAO.create method here. No need to set the ID manually.

File: src/main/java/terminallayer/TerminalMain.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+                BigDecimal deposit = scan.nextBigDecimal();
+                accountsDAO.makeDeposit(a, deposit);
+                break;
+            case 3:
+                makeTransfer(a);
+                break;
+            case 4:
+                LOGGER.info("Amount to Withdraw");
+                BigDecimal withdraw = scan.nextBigDecimal();
+                accountsDAO.makeWithdrawal(a, withdraw);
```
Comment (Moussa-Mikhail) [inline src/main/java/terminallayer/TerminalMain.java]: For both makeTransfer and makeWithdrawal you should catch the DAOException (which is thrown when an account has insufficient funds) and inform the user of the issue.

**Comment by Moussa-Mikhail** [comment]
Everything looks good to me. Thanks for making those changes and good catch on generateCvc. Merging.

**Pull request merged**